src/main/java/com/ericsson/ei/subscription/SubscriptionValidator.java

```diff
@@ -59,127 +60,143 @@ private SubscriptionValidator() {
-     * wrong format of parameter.
+     * Validation of NotificationMessageKeyValues parameters (key/values).
+     * Throws SubscriptionValidationException if validation of the parameter
+     * fails due to wrong format of parameter.
      *
-     * @param notificationMessage
-     *     The message body to use for a notification of a subscription
-     * @param restPostBodyMediaType
-     *     The media type of the REST POST request body
+     * @param notificationMessage   The message body to use for a notification
+     *                              of a subscription
+     * @param restPostBodyMediaType The media type of the REST POST request body
      * @throws SubscriptionValidationException
      */
-    private static void validateNotificationMessageKeyValues(List<NotificationMessageKeyValue> notificationMessage,
-            String restPostBodyMediaType) throws SubscriptionValidationException {
-        for (NotificationMessageKeyValue item : notificationMessage) {
-            String testKey = item.getFormkey();
-            String testValue = item.getFormvalue();
-            if (restPostBodyMediaType != null
-                    && restPostBodyMediaType.equals(MediaType.APPLICATION_FORM_URLENCODED.toString())) { // FORM/POST
-                // PARAMETERS
-                if (StringUtils.isBlank(testKey) || StringUtils.isBlank(testValue)) {
-                    throw new SubscriptionValidationException(
-                            "Value & Key  in notificationMessage must have a values: " + notificationMessage);
-                }
-            } else {
-                if (notificationMessage.size() != 1) {
-                    throw new SubscriptionValidationException(
-                            "Only one array is allowed for notificationMessage when NOT using key/value pairs: "
-                                    + notificationMessage);
-                } else if (testKey != null && !testKey.isEmpty()) {
-                    throw new SubscriptionValidationException(
-                            "Key in notificationMessage must be empty when NOT using key/value pairs: "
-                                    + notificationMessage);
-                } else if (StringUtils.isBlank(testValue)) {
-                    throw new SubscriptionValidationException(
-                            "Value in notificationMessage must have a value when NOT using key/value pairs: "
-                                    + notificationMessage);
-                }
-            }
-        }
+    private static void validateNotificationMessageKeyValues(
+	    List<NotificationMessageKeyValue> notificationMessage,
+	    String restPostBodyMediaType)
+	    throws SubscriptionValidationException {
+	for (NotificationMessageKeyValue item : notificationMessage) {
+	    String testKey = item.getFormkey();
+	    String testValue = item.getFormvalue();
+	    if (restPostBodyMediaType != null && restPostBodyMediaType
```

Comment:
Please move this statement to its own variable

Reply:
Done.